### ISSUE TITLE
Showing quota of webdav remotes(e.g. nextcloud)

### ISSUE THREAD

**Issue by homeii** [ncw/rclone#2937]
#### What is your current rclone version (output from `rclone version`)?

rclone v1.45-108-g184a9c8d-beta

#### What problem are you are trying to solve?

Showing the real used&free space of webdav mount.

#### How do you think rclone should be changed to solve that?

[Rfc4331](http://www.webdav.org/specs/rfc4331.html) allowed clients to query quota infomations of the webdav server.
Both Nextcloud and Owncloud supports it.
Tested with curl:
```bash
curl -u admin:admin 'https://NEXTCLOUD_URL/remote.php/webdav/' -X PROPFIND --data '<?xml version="1.0" ?><D:propfind xmlns:D="DAV:"><D:prop>           <D:quota-available-bytes/><D:quota-used-bytes/></D:prop></D:propfind>' -H "Depth: 0"
```
Additional header:
```
Depth: 0
```
Post data:
```xml
      <?xml version="1.0" ?>
      <D:propfind xmlns:D="DAV:">
        <D:prop>
          <D:quota-available-bytes/>
          <D:quota-used-bytes/>
        </D:prop>
      </D:propfind>
```
Response:
```xml
<?xml version="1.0"?> 
<d:multistatus xmlns:d="DAV:" xmlns:s="http://sabredav.org/ns" xmlns:oc="http://owncloud.org/ns" xmlns:nc="http://nextcloud.org/ns">
    <d:response>
        <d:href>/remote.php/webdav/</d:href>
        <d:propstat>
               <d:prop>
                    <d:quota-available-bytes>93889607</d:quota-available-bytes>
                    <d:quota-used-bytes>10967993</d:quota-used-bytes>
                </d:prop>
               <d:status>HTTP/1.1 2 00 OK</d:status>
        </d:propstat>
    </d:response>
</d:multistatus>
```

**Comment by ncw** [comment]
When I try this on my nextcloud and owncloud I get `-3` for `quota-available-bytes` - have you seen an explanation for that anywhere?

**Comment by ncw** [comment]
I had an initial go at this - tell me what you think?

https://beta.rclone.org/branch/v1.45-110-gff8c2d67-fix-2379-owncloud-hashes-beta/ (uploaded in 15-30 mins)

**Comment by DigitalJosee** [comment]
Hey, I've this same problem, is only me or this link isn't working?

**Comment by homeii** [comment]
> I had an initial go at this - tell me what you think?
> 
> https://beta.rclone.org/branch/v1.45-110-gff8c2d67-fix-2379-owncloud-hashes-beta/ (uploaded in 15-30 mins)

It seems that this beta version works well with nextcloud both when the quota is set or not.I also tested it with SabreDAV ,it also works.

![image](https://user-images.githubusercontent.com/7547189/51815198-4bc6d100-22fa-11e9-8887-6e11418d5137.png)


![image](https://user-images.githubusercontent.com/7547189/51815192-423d6900-22fa-11e9-8772-20a888157d81.png)

**Comment by ncw** [comment]
> So when you set your quota to `unlimited` in nextcloud(like this), you'll get a response with `quota-available-bytes=-3`.

Thanks for working that out.  What I've done is assumed that if the quote bytes are -ve they are unlimited.  This fits into rclone's framework very well.

Thanks for testing

I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.46
